### PR TITLE
feat: upgrade-nx-14

### PR DESCRIPTION
**Pull request by ryanbas21**
# JIRA Ticket

n/a

# Description
upgrades infra to nx-14

# Type of Change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

CI and locally running apps

# Definition of Done

Check all that apply

- [ ] Acceptance criteria is met.
- [ ] All tasks listed in the user story have been completed.
- [ ] Coded to standards.
- [ ] Code peer-reviewed.
- [ ] Ensure backward compatibility (special attention).
- [ ] API reference docs is updated.
- [ ] Unit tests are written.
- [ ] Integration tests are written.
- [ ] e2e tests are written.
- [ ] CI build passing on the feature branch.
- [ ] Functional spec is written/updated
- [ ] contains example code snippets.
- [ ] Change log updated.
- [ ] Documentation story is created and tracked.
- [ ] UI is completed or ticket is created.
- [ ] Demo to PO and team.
- [ ] Tech debts and remaining tasks are tracked in separated ticket(s).

## Documentation

- [ ] Acceptance criteria met
- [ ] Spell-check run
- [ ] Peer reviewed
- [ ] Proofread

### CODE REVIEW

File: migrations.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{
```
Comment (cerebrl) [inline migrations.json:1]: What's the purpose of this file?

File: packages/javascript-sdk/jest.basic.config.ts [hunk 2]
```diff
@@ -7,10 +7,10 @@
  * This software may be modified and distributed under the terms
  * of the MIT license. See the LICENSE file for details.
  */
-const config = require('./jest.config');
+import config from './jest.config';
 
-module.exports = {
-  ...config,
+export default {
+  // ...config,
```
Comment (cerebrl) [inline packages/javascript-sdk/jest.basic.config.ts:13]: If we are no longer spreading `config` into the exported object here, do we need to import it?

Comment (ryanbas21) [reply packages/javascript-sdk/jest.basic.config.ts:13]: ah good catch, we need this. i had to comment these out to make the upgrade work, so i have to comment it back in.

File: migrations.json (hunk 1 above)
Comment (ryanbas21) [reply migrations.json:1]: should delete and ill add this to gitignore, its a file nx generates about what the migration is, and then you run-migrations with this file

File: packages/javascript-sdk/jest.basic.config.ts [hunk 3]
```diff
@@ -7,9 +7,9 @@
  * This software may be modified and distributed under the terms
  * of the MIT license. See the LICENSE file for details.
  */
-const config = require('./jest.config');
+import config from './jest.config';
```
Comment (cerebrl) [inline packages/javascript-sdk/jest.basic.config.ts:10]: This is awesome!

**Review by cerebrl — APPROVED** [review]

**Pull request merged**